cranelift/docs/isle.md

```diff
@@ -0,0 +1,1306 @@
+# ISLE: Instruction Selection Lowering Expressions
```

Comment:
Ah, OK, I was following your PR's lead (`cranelift/docs/isle-integration.md`) but I've moved both this and the ISLE-Cranelift integration doc into `cranelift/isle/docs/`. (Also renamed `isle-integration.md` to `cranelift-integration.md` since it's now within the ISLE tree.) This is nicer, agreed!

Reply:
Wait, but the `isle-integration.md` stuff *is* Cranelift-specific, so it does make sense to be inside `cranelift/docs`.

Where as this documentation is just about ISLE and not any specific embedding, so it makes sense it should be under `cranelift/isle/docs` so it is a part of the ISLE crate/package.

Reply:
Ah, two different perspectives I guess. I had figured putting all ISLE-related documentation in one place made sense, otherwise a newcomer who just knows "ISLE is an isel thing" and goes looking for its documentation might miss some of it. I don't care too much either way though, so I'm happy to move it back.